src/reverse_complement.rs

```diff
@@ -10,12 +10,11 @@ extern crate rayon;
 extern crate memchr;
 
 use std::io::{Read, Write};
-use std::{io, ptr, slice};
+use std::{cmp, io, mem};
 use std::fs::File;
 
 struct Tables {
```

Comment:
The last one must generate bound check, so better having a `[u8; 256]` to elide bound checks.